File: Projet_final/angularBT/src/app/model/projecteur.ts
import { Cursus } from './cursus';
import { Materiel } from './materiel';
export class Projecteur extends Materiel{
  constructor(private _cursus: Cursus) {
    super();
  }
  public get cursus(): Cursus {
    return this._cursus;
  }
   public set cursus(_cursus: Cursus) {
    this._cursus = _cursus;
  }
}

File: Projet_final/projetFinalSpringBoot/src/main/java/com/exemple/model/Projecteur.java
package com.exemple.model;

import javax.persistence.DiscriminatorValue;
import javax.persistence.Entity;
import javax.persistence.JoinColumn;
import javax.persistence.OneToOne;

import com.fasterxml.jackson.annotation.JsonView;

@Entity
@DiscriminatorValue("Video_projecteur")
public class Projecteur extends Materiel {
	@OneToOne
	@JoinColumn(name = "Projecteur_cursus")
	@JsonView(JsonViews.Materiel.class)
	private Cursus cursus;

	public Cursus getCursus() {
		return cursus;
	}

	public void setCursus(Cursus cursus) {
		this.cursus = cursus;
	}
}
